algorithms/merge-sort.js

```diff
@@ -1 +1,33 @@
+function mergeSort(arr) {
+  let lHalf = arr.slice(0, arr.length / 2);
+  let rHalf = arr.slice(arr.length / 2, arr.length);
+  sort(lHalf);
+  sort(rHalf);
+  arr = merge(lHalf, rHalf);
+  return arr;
+}
 
+function sort(arr) {
+  let temp;
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr.length; j++) {
+      if (arr[i] < arr[j]) {
+        temp = arr[i];
+        arr[i] = arr[j];
+        arr[j] = temp;
+      }
+    }
+  }
```

Comment:
The resetting of both indices, `i` and `j` to 0 on cycle is a time waste. Merge sort assumes the `left` and `right` coming in are sorted themselves. Using that to our advantage, we only have to go to each array once. Which becomes `2n` rather than `n^2`.